src/main/java/net/neoforged/neoforge/common/extensions/IMobEffectInstanceExtension.java

```diff
@@ -1,56 +0,0 @@
-/*
- * Copyright (c) Forge Development LLC and contributors
- * SPDX-License-Identifier: LGPL-2.1-only
- */
-
-package net.neoforged.neoforge.common.extensions;
-
-import java.util.List;
-import net.minecraft.nbt.CompoundTag;
-import net.minecraft.nbt.ListTag;
-import net.minecraft.world.effect.MobEffect;
-import net.minecraft.world.item.ItemStack;
-
-public interface IMobEffectInstanceExtension {
-
-    /***
-     * Returns a list of curative items for the potion effect
-     * By default, this list is initialized using {@link MobEffect#getCurativeItems()}
-     *
-     * @return The list (ItemStack) of curative items for the potion effect
-     */
-    List<ItemStack> getCurativeItems();
-
-    /***
-     * Checks the given ItemStack to see if it is in the list of curative items for the potion effect
-     * 
-     * @param stack The ItemStack being checked against the list of curative items for this PotionEffect
-     * @return true if the given ItemStack is in the list of curative items for this PotionEffect, false otherwise
-     */
-    default boolean isCurativeItem(ItemStack stack) {
-        return this.getCurativeItems().stream().anyMatch(e -> ItemStack.isSameItem(e, stack));
-    }
-
-    /***
-     * Sets the list of curative items for this potion effect, overwriting any already present
-     * 
-     * @param curativeItems The list of ItemStacks being set to the potion effect
-     */
-    void setCurativeItems(List<ItemStack> curativeItems);
-
-    /***
-     * Adds the given stack to the list of curative items for this PotionEffect
-     * 
-     * @param stack The ItemStack being added to the curative item list
-     */
-    default void addCurativeItem(ItemStack stack) {
-        if (!this.isCurativeItem(stack))
-            this.getCurativeItems().add(stack);
-    }
-
-    default void writeCurativeItems(CompoundTag nbt) {
```

Comment:
I do wonder if its worth still using this for `readCures` and `writeCures`, we can still clear/add to the set via `getCures()` as its not an immutable wrapper. Its a small patch either way, but guess its a patch into fewer places without the extension interface.

Reply:
I don't see much point in keeping the extension interface in this case because the patch is so small anyway.